### PR TITLE
Fixing page limit on cosmosdb

### PR DESCRIPTION
**Pull request by RyanLettieri**
Signed-off-by: Ryan Lettieri <ryanLettieri@microsoft.com>

# Description
Fixed page size limit bug in cosmosdb statestore

## Issue reference

No issue to reference

### CODE REVIEW

File: state/azure/cosmosdb/cosmosdb_query.go [hunk 1]
```diff
@@ -147,9 +147,10 @@ func (q *Query) setNextParameter(val string) string {
 func (q *Query) execute(client *azcosmos.ContainerClient) ([]state.QueryItem, string, error) {
 	opts := &azcosmos.QueryOptions{}
 
+	resultLimit := q.limit
 	opts.QueryParameters = append(opts.QueryParameters, q.query.parameters...)
 	if q.limit != 0 {
-		opts.PageSizeHint = int32(q.limit)
+		opts.PageSizeHint = int32(resultLimit)
```
Comment (berndverst) [inline state/azure/cosmosdb/cosmosdb_query.go:153]: Let's not set the Page Size at all - let's rely on the default. We iterate through all pages anyway to fetch all results.

**Review by berndverst — CHANGES_REQUESTED** [review]

**Review by berndverst — DISMISSED** [review]

File: state/azure/cosmosdb/cosmosdb_query.go [hunk 2]
```diff
@@ -175,8 +174,14 @@ func (q *Query) execute(client *azcosmos.ContainerClient) ([]state.QueryItem, st
 			if err != nil {
 				return nil, "", err
 			}
+			if (len(items) + 1) > resultLimit {
+				break
+			}
 			items = append(items, tempItem)
 		}
+		if (len(items) + 1) > resultLimit {
```
Comment (artursouza) [inline state/azure/cosmosdb/cosmosdb_query.go:182]: nit: can this be a simple ">="?

File: state/azure/cosmosdb/cosmosdb_query.go [hunk 3]
```diff
@@ -175,8 +174,14 @@ func (q *Query) execute(client *azcosmos.ContainerClient) ([]state.QueryItem, st
 			if err != nil {
 				return nil, "", err
 			}
+			if (len(items) + 1) > resultLimit {
```
Comment (artursouza) [inline state/azure/cosmosdb/cosmosdb_query.go:177]: >= ?

**Review by berndverst — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Review by berndverst — CHANGES_REQUESTED** [review]
Tests failing. 👀

File: state/azure/cosmosdb/cosmosdb_query.go [hunk 4]
```diff
@@ -175,8 +174,14 @@ func (q *Query) execute(client *azcosmos.ContainerClient) ([]state.QueryItem, st
 			if err != nil {
 				return nil, "", err
 			}
+			if len(items) >= resultLimit {
```
Comment (berndverst) [inline state/azure/cosmosdb/cosmosdb_query.go:177]:
```suggestion
			if (resultLimit != 0) && (len(items) >= resultLimit) {
```

File: state/azure/cosmosdb/cosmosdb_query.go [hunk 5]
```diff
@@ -175,8 +174,14 @@ func (q *Query) execute(client *azcosmos.ContainerClient) ([]state.QueryItem, st
...
+				break
+			}
 			items = append(items, tempItem)
 		}
+		if len(items) >= resultLimit {
```
Comment (berndverst) [inline state/azure/cosmosdb/cosmosdb_query.go:182]:
```suggestion
		if (resultLimit != 0) && (len(items) >= resultLimit) {
```

**Review by berndverst — APPROVED** [review]
Tests failing. 👀

**Pull request merged**